From dc66021677082317f64907d6307c2c6d298dc648 Mon Sep 17 00:00:00 2001
From: Goffin Simon <sig@odoo.com>
Date: Wed, 8 May 2019 08:24:37 +0000
Subject: [PATCH] [FIX] l10n_be_reports: Partner VAT Intra

When generating the report, partners with the same VAT Number and same code were on separate lines.
But in Partner VAT listing, the partners with the same VAT number are summed.

Forward-port up to saas-15

opw:1972674

closes odoo/enterprise#4286

Signed-off-by: Simon Goffin (sig) <sig@openerp.com>
---
 l10n_be_reports/models/partner_vat_intra.py | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/l10n_be_reports/models/partner_vat_intra.py b/l10n_be_reports/models/partner_vat_intra.py
index 0ce075d..9a53502 100644
--- a/l10n_be_reports/models/partner_vat_intra.py
+++ b/l10n_be_reports/models/partner_vat_intra.py
@@ -18,7 +18,14 @@ class ReportL10nBePartnerVatIntra(models.AbstractModel):
             company_ids = '(' + ','.join(map(str, context_id.company_ids.ids)) + ')'
             company_clause = 'AND l.company_id IN ' + company_ids
         tag_ids = [self.env['ir.model.data'].xmlid_to_res_id(k) for k in ['l10n_be.tax_tag_44', 'l10n_be.tax_tag_46L', 'l10n_be.tax_tag_46T']]
-        self.env.cr.execute('''SELECT p.name As partner_name, l.partner_id AS partner_id, p.vat AS vat,
+        if get_xml_data:
+            group_by = 'p.vat, intra_code'
+            select = ''
+        else:
+            group_by = 'p.name, l.partner_id, p.vat, intra_code'
+            select = 'p.name As partner_name, l.partner_id AS partner_id,'
+        query = """
+        SELECT {select} p.vat AS vat,
                       tt.account_account_tag_id AS intra_code, SUM(-l.balance) AS amount
                       FROM account_move_line l
                       LEFT JOIN res_partner p ON l.partner_id = p.id
@@ -28,8 +35,9 @@ class ReportL10nBePartnerVatIntra(models.AbstractModel):
                        AND l.date >= '%s'
                        AND l.date <= '%s'
                        %s
-                      GROUP BY p.name, l.partner_id, p.vat, intra_code''' % (tuple(tag_ids), context_id.date_from, context_id.date_to, company_clause))
-
+                      GROUP BY {group_by}
+        """ % (tuple(tag_ids), context_id.date_from, context_id.date_to, company_clause)
+        self.env.cr.execute(query.format(select=select, group_by=group_by))
         p_count = 0
 
         for row in self.env.cr.dictfetchall():
@@ -50,10 +58,10 @@ class ReportL10nBePartnerVatIntra(models.AbstractModel):
                     columns[3] = formatLang(self.env, columns[3], currency_obj=currency_id)
 
                 lines.append({
-                    'id': row['partner_id'],
+                    'id': row['partner_id'] if not get_xml_data else False,
                     'type': 'partner_id',
-                    'name': row['partner_name'],
-                    'footnotes': context_id._get_footnotes('partner_id', row['partner_id']),
+                    'name': row['partner_name'] if not get_xml_data else False,
+                    'footnotes': context_id._get_footnotes('partner_id', row['partner_id']) if not get_xml_data else False,
                     'columns': columns,
                     'level': 2,
                     'unfoldable': False,
-- 
2.7.4

